### ISSUE TITLE
Sound not working on macOS

### ISSUE THREAD

**Issue by fhughes90** [hjdhjd/homebridge-unifi-protect#71]
Viewing live video in Homekit on iOS works great, but I noticed I do not have sound when I am using my Mac to view Homekit.

My setup:
Running homebridge on RPi4 4GB model

**Comment by hjdhjd** [comment]
Thanks for the report. Unfortunately, this isn't a particularly complete bug report and I can't help much without a lot more information. What OS versions are you running? Protect versions? Plugin version?

I don't believe this is a plugin issue, for what it's worth. I've got the Home app running on Catalina and Mojave without an issue, including sound.

**Comment by fhughes90** [comment]
Your right, my apologies. I can also send the verbose ffmpeg logging, I just didnt know what distinguishes an error on the sound in the logging.

Here is the information you asked for:

macOS version: Catalina
Protect controller version: 1.13.4
Protect Hardware: CK2+
Homebridge plug-in: homebridge-Unifi-protect v3.6.4

**Comment by hjdhjd** [comment]
No worries. We’re all trying to help here. I think you may have a version of ffmpeg that doesn’t support audio. How exactly did you install the plugin? The version of ffmpeg that the plugin will try to use comes from ffmpeg-for-homebridge, which helps to streamline this whole process. However, if you didn’t install the plugin with —unsafe-perm specified, you can end up using whatever default ffmpeg you might have already on your system, and not one that has fdk_aac support. :smile:

So...how’d you install the plugin exactly?

Also...feel free to post the logs from the plugin...it produces a fair amount of logging, on it’s own and usually tells you what’s going on.

Thanks.

**Comment by fhughes90** [comment]
Yeah, that’s a good assumption. I use to run my Protect cameras through homebridge utilizing the plugin homebridge-camera-ffmpeg, and I entered the config manually for each camera and with that I compiled ffmpeg with all the binaries manually.

Since this amazing plugin has surfaced, I switched to it and followed the installation with these two commands for ffmpeg:

`sudo npm install -g homebridge-camera-ffmpeg --unsafe-perm`

`sudo curl -Lf# https://github.com/homebridge/ffmpeg-for-homebridge/releases/latest/download/ffmpeg-raspbian-armv6l.tar.gz | sudo tar xzf - -C / --no-same-owner `

**Comment by hjdhjd** [comment]
No stop! :smile:

Do **not** install ffmpeg-for-homebridge directly. The plugins, both mine and `homebridge-camera-ffmpeg` will install the dependency automatically under our own namespaces.

Do this:

BTW...I don't get why you're doing any of this as root on a Mac, but to each their own...it's unnecessary, FYI.

1. Stop homebridge.
2. Uninstall the `homebridge-unifi-protect` plugin.
3. Install the plugin by entering:
```
npm -g --unsafe-perm homebridge-unifi-protect
```
Slap a `sudo` on that if you're doing all this as root.

4. Start homebridge.

Give that a shot. Also, make sure you are using only the default settings in the plugin to get up and running.

**Comment by fhughes90** [comment]
Alright, I tried your recommendations and I seem to have the same symptoms - sound comes through on iOS but now on my macOS. I'm almost tempted to reformat the RPi and give it a fresh start (I monkey'ed around to get FDK-AAC and FFMPEG working on the RPi before I knew of these plugins)

```
[9/9/2020, 10:15:45] [UniFi Protect] Hughes NVR [UCK-G2-PLUS] Front Camera [UVC G3]: HomeKit video stream request received: 1280x720, 30 fps, 299 kbps.
[9/9/2020, 10:15:45] [UniFi Protect] Hughes NVR [UCK-G2-PLUS] Front Camera [UVC G3]: ffmpeg command: /usr/lib/node_modules/homebridge-unifi-protect/node_modules/ffmpeg-for-homebridge/ffmpeg [
  '-hide_banner',
  '-rtsp_transport',
  'tcp',
  '-i',
  'rtsp://192.168.1.2:7447/9oxWApqFxWgMeFD9',
  '-map',
  '0:v',
  '-vcodec',
  'copy',
  '-f',
  'rawvideo',
  '-pix_fmt',
  'yuvj420p',
  '-r',
  '30',
  '-probesize',
  '32',
  '-analyzeduration',
  '0',
  '-fflags',
  'nobuffer',
  '-b:v',
  '299k',
  '-bufsize',
  '598k',
  '-maxrate',
  '299k',
  '-payload_type',
  '99',
  '-ssrc',
  '14797017',
  '-f',
  'rtp',
  '-srtp_out_suite',
  'AES_CM_128_HMAC_SHA1_80',
  '-srtp_out_params',
  'gqkbseeyZ4hHxStk2nsWGwVCzFXhu1rIxmj629z1',
  'srtp://192.168.2.51:57198?rtcpport=57198&localrtcpport=57198&pkt_size=564',
  '-map',
  '0:a',
  '-acodec',
  'libfdk_aac',
  '-profile:a',
  'aac_eld',
  '-flags',
  '+global_header',
  '-f',
  'null',
  '-ar',
  '16k',
  '-b:a',
  '24k',
  '-bufsize',
  '48k',
  '-ac',
  '1',
  '-payload_type',
  '110',
  '-ssrc',
  '13747277',
  '-f',
  'rtp',
  '-srtp_out_suite',
  'AES_CM_128_HMAC_SHA1_80',
  '-srtp_out_params',
  'cd2mRuF3x693e3qdw8cODW99qvt7xOFqRUSyFFX5',
  'srtp://192.168.2.51:55484?rtcpport=55484&localrtcpport=55484&pkt_size=188',
  '-loglevel',
  'level+verbose'
]
[9/9/2020, 10:15:45] [UniFi Protect] [tcp @ 0x3d44af0] [verbose] Starting connection attempt to XXX.XXX.XXX.XXX port 7447
[9/9/2020, 10:15:45] [UniFi Protect]
[9/9/2020, 10:15:45] [UniFi Protect] [tcp @ 0x3d44af0] [verbose] Successfully connected to XXX.XXX.XXX.XXX port 7447
[9/9/2020, 10:15:45] [UniFi Protect]
[9/9/2020, 10:15:45] [UniFi Protect] [rtsp @ 0x3d42ab0] [verbose] SDP:
[9/9/2020, 10:15:45] [UniFi Protect] v=0
[9/9/2020, 10:15:45] [UniFi Protect] o=- 681 0 IN IP4 192.168.1.2
[9/9/2020, 10:15:45] [UniFi Protect] s=FCECDAD92F19_0
[9/9/2020, 10:15:45] [UniFi Protect] u=www.evostream.com
[9/9/2020, 10:15:45] [UniFi Protect] e=contact@evostream.com
[9/9/2020, 10:15:45] [UniFi Protect] c=IN IP4 192.168.1.2
[9/9/2020, 10:15:45] [UniFi Protect] t=0 0
[9/9/2020, 10:15:45] [UniFi Protect] a=recvonly
[9/9/2020, 10:15:45] [UniFi Protect] a=control:*
[9/9/2020, 10:15:45] [UniFi Protect] a=range:npt=now-
[9/9/2020, 10:15:45] [UniFi Protect] m=audio 0 RTP/AVP 96
[9/9/2020, 10:15:45] [UniFi Protect] a=recvonly
[9/9/2020, 10:15:45] [UniFi Protect] a=rtpmap:96 mpeg4-generic/11025/1
[9/9/2020, 10:15:45] [UniFi Protect] a=control:trackID=1
[9/9/2020, 10:15:45] [UniFi Protect] a=fmtp:96 streamtype=5; profile-level-id=15; mode=AAC-hbr; config=1508; SizeLength=13; IndexLength=3; IndexDeltaLength=3;
[9/9/2020, 10:15:45] [UniFi Protect] m=video 0 RTP/AVP 97
[9/9/2020, 10:15:45] [UniFi Protect] a=recvonly
[9/9/2020, 10:15:45] [UniFi Protect] a=control:trackID=2
[9/9/2020, 10:15:45] [UniFi Protect] a=rtpmap:97 H264/90000
[9/9/2020, 10:15:45] [UniFi Protect] a=fmtp:97 profile-level-id=4d0029; packetization-mode=1; sprop-parameter-sets=Z00AKZpkA8ARPy/+AXoBfNwEBAUAAAMD6AAAdTCdoIhGoA==,aO48gA==
[9/9/2020, 10:15:45] [UniFi Protect]
[9/9/2020, 10:15:45] [UniFi Protect] [rtsp @ 0x3d42ab0] [verbose] setting jitter buffer size to 0
[9/9/2020, 10:15:45] [UniFi Protect]
[9/9/2020, 10:15:45] [UniFi Protect] [rtsp @ 0x3d42ab0] [verbose] setting jitter buffer size to 0
[9/9/2020, 10:15:45] [UniFi Protect]
[9/9/2020, 10:15:45] [UniFi Protect] [h264 @ 0x3d48fd0] [verbose] Reinit context to 1920x1088, pix_fmt: yuvj420p
[9/9/2020, 10:15:45] [UniFi Protect]
[9/9/2020, 10:15:47] [UniFi Protect] [info]
[9/9/2020, 10:15:47] [UniFi Protect] Input #0, rtsp, from 'rtsp://192.168.1.2:7447/9oxWApqFxWgMeFD9':
[9/9/2020, 10:15:47] [UniFi Protect]
[9/9/2020, 10:15:47] [UniFi Protect] [info]
[9/9/2020, 10:15:47] [UniFi Protect]   Metadata:
[9/9/2020, 10:15:47] [UniFi Protect]
[9/9/2020, 10:15:47] [UniFi Protect] [info]
[9/9/2020, 10:15:47] [UniFi Protect]     title           : FCECDAD92F19_0
[9/9/2020, 10:15:47] [UniFi Protect] [info]   Duration: N/A, start: 0.000000, bitrate: N/A
[9/9/2020, 10:15:47] [UniFi Protect] [info]     Stream #0:0: Audio: aac (LC), 11025 Hz, mono, fltp
[9/9/2020, 10:15:47] [UniFi Protect] [info]     Stream #0:1: Video: h264 (Main), 1 reference frame, yuvj420p(pc, bt709, progressive, left), 1920x1080 (1920x1088) [SAR 189:190 DAR 168:95], 15 fps, 15 tbr, 90k tbn, 30 tbc
[9/9/2020, 10:15:47] [UniFi Protect]
[9/9/2020, 10:15:47] [UniFi Protect] [info] Output #0, rtp, to 'srtp://192.168.2.51:57198?rtcpport=57198&localrtcpport=57198&pkt_size=564':
[9/9/2020, 10:15:47] [UniFi Protect] [info]   Metadata:
[9/9/2020, 10:15:47] [UniFi Protect] [info]     title           : FCECDAD92F19_0
[9/9/2020, 10:15:47] [UniFi Protect] [info]     encoder         : Lavf58.45.100
[9/9/2020, 10:15:47] [UniFi Protect] [info]     Stream #0:0: Video: h264 (Main), 1 reference frame, yuvj420p(pc, bt709, progressive, left), 1920x1080 (0x0) [SAR 189:190 DAR 168:95], q=2-31, 299 kb/s, 15 fps, 15 tbr, 90k tbn, 30 tbc
[9/9/2020, 10:15:47] [UniFi Protect] [info] Stream mapping:
[9/9/2020, 10:15:47] [UniFi Protect] [info]   Stream #0:1 -> #0:0 (copy)
[9/9/2020, 10:15:47] [UniFi Protect] [info]   Stream #0:0 -> #1:0 (aac (native) -> aac (libfdk_aac))
[9/9/2020, 10:15:47] [UniFi Protect] [info] Press [q] to stop, [?] for help
[9/9/2020, 10:15:47] [UniFi Protect]
[9/9/2020, 10:15:47] [UniFi Protect] [graph_0_in_0_0 @ 0x3e660c0] [verbose] tb:1/11025 samplefmt:fltp samplerate:11025 chlayout:0x4
[9/9/2020, 10:15:47] [UniFi Protect] [format_out_1_0 @ 0x3e66370] [verbose] auto-inserting filter 'auto_resampler_0' between the filter 'Parsed_anull_0' and the filter 'format_out_1_0'
[9/9/2020, 10:15:47] [UniFi Protect]
[9/9/2020, 10:15:47] [UniFi Protect] [auto_resampler_0 @ 0x3d4f2f0] [verbose] ch:1 chl:mono fmt:fltp r:11025Hz -> ch:1 chl:mono fmt:s16 r:16000Hz
[9/9/2020, 10:15:47] [UniFi Protect] [info] Output #1, rtp, to 'srtp://192.168.2.51:55484?rtcpport=55484&localrtcpport=55484&pkt_size=188':
[9/9/2020, 10:15:47] [UniFi Protect] [info]   Metadata:
[9/9/2020, 10:15:47] [UniFi Protect] [info]     title           : FCECDAD92F19_0
[9/9/2020, 10:15:47] [UniFi Protect] [info]     encoder         : Lavf58.45.100
[9/9/2020, 10:15:47] [UniFi Protect] [info]     Stream #1:0: Audio: aac (libfdk_aac) (ELD), 16000 Hz, mono, s16, delay 288, 24 kb/s
[9/9/2020, 10:15:47] [UniFi Protect] [info]     Metadata:
[9/9/2020, 10:15:47] [UniFi Protect] [info]       encoder         : Lavc58.91.100 libfdk_aac
[9/9/2020, 10:15:47] [UniFi Protect]
[9/9/2020, 10:15:47] [UniFi Protect] [info] frame=   36 fps=0.0 q=-1.0 size=     373kB time=00:00:02.47 bitrate=1232.0kbits/s speed=4.78x
[9/9/2020, 10:15:48] [UniFi Protect] [info] frame=   44 fps= 42 q=-1.0 size=     403kB time=00:00:03.02 bitrate=1092.4kbits/s speed=2.87x
[9/9/2020, 10:15:48] [UniFi Protect] [info] frame=   52 fps= 33 q=-1.0 size=     430kB time=00:00:03.59 bitrate= 978.8kbits/s speed=2.27x
[9/9/2020, 10:15:49] [UniFi Protect] [info] frame=   61 fps= 29 q=-1.0 size=     457kB time=00:00:04.06 bitrate= 920.2kbits/s speed=1.92x
[9/9/2020, 10:15:49] [UniFi Protect] [info] frame=   68 fps= 26 q=-1.0 size=     480kB time=00:00:04.62 bitrate= 851.5kbits/s speed=1.74x
[9/9/2020, 10:15:50] [UniFi Protect] [info] frame=   75 fps= 24 q=-1.0 size=     504kB time=00:00:05.07 bitrate= 815.0kbits/s speed=1.61x
[9/9/2020, 10:15:50] [UniFi Protect] [info] frame=   84 fps= 23 q=-1.0 size=     756kB time=00:00:05.64 bitrate=1097.3kbits/s speed=1.54x
[9/9/2020, 10:15:51] [UniFi Protect] [info] frame=   91 fps= 22 q=-1.0 size=     777kB time=00:00:06.19 bitrate=1028.0kbits/s speed=1.48x
[9/9/2020, 10:15:51] [UniFi Protect] [info] frame=  100 fps= 21 q=-1.0 size=     804kB time=00:00:06.66 bitrate= 988.4kbits/s speed=1.41x
[9/9/2020, 10:15:52] [UniFi Protect] [info] frame=  107 fps= 20 q=-1.0 size=     824kB time=00:00:07.21 bitrate= 935.3kbits/s speed=1.37x
[9/9/2020, 10:15:52] [UniFi Protect] [info] frame=  115 fps= 20 q=-1.0 size=     848kB time=00:00:07.75 bitrate= 895.2kbits/s speed=1.34x
[9/9/2020, 10:15:53] [UniFi Protect] [info] frame=  123 fps= 19 q=-1.0 size=     873kB time=00:00:08.33 bitrate= 858.5kbits/s speed=1.32x
[9/9/2020, 10:15:53] [UniFi Protect] [info] frame=  132 fps= 19 q=-1.0 size=     903kB time=00:00:08.80 bitrate= 840.6kbits/s speed=1.28x
[9/9/2020, 10:15:54] [UniFi Protect] [info] frame=  139 fps= 19 q=-1.0 size=     928kB time=00:00:09.35 bitrate= 812.3kbits/s speed=1.27x
[9/9/2020, 10:15:55] [UniFi Protect] [info] frame=  147 fps= 19 q=-1.0 size=     952kB time=00:00:09.90 bitrate= 787.3kbits/s speed=1.25x
[9/9/2020, 10:15:55] [UniFi Protect] [info] frame=  156 fps= 18 q=-1.0 size=    1208kB time=00:00:10.40 bitrate= 951.7kbits/s speed=1.23x
[9/9/2020, 10:15:56] [UniFi Protect] [info] frame=  163 fps= 18 q=-1.0 size=    1227kB time=00:00:10.92 bitrate= 920.3kbits/s speed=1.22x
[9/9/2020, 10:15:56] [UniFi Protect] [info] frame=  171 fps= 18 q=-1.0 size=    1254kB time=00:00:11.47 bitrate= 895.3kbits/s speed=1.21x
[9/9/2020, 10:15:57] [UniFi Protect] [info] frame=  179 fps= 18 q=-1.0 size=    1279kB time=00:00:12.04 bitrate= 869.9kbits/s speed= 1.2x
[9/9/2020, 10:15:57] [UniFi Protect] [info] frame=  188 fps= 18 q=-1.0 size=    1305kB time=00:00:12.53 bitrate= 852.7kbits/s speed=1.18x
[9/9/2020, 10:15:58] [UniFi Protect] [info] frame=  195 fps= 18 q=-1.0 size=    1329kB time=00:00:13.07 bitrate= 833.0kbits/s speed=1.18x
[9/9/2020, 10:15:58] [UniFi Protect] [info] frame=  203 fps= 17 q=-1.0 size=    1358kB time=00:00:13.61 bitrate= 817.4kbits/s speed=1.17x
[9/9/2020, 10:15:58] [UniFi Protect] Hughes NVR [UCK-G2-PLUS] Front Camera [UVC G3]: Stopped video streaming session.
```

**Comment by hjdhjd** [comment]
Everything looks to be working as intended. If you aren't getting audio on your Mac, but are on iOS, I would strongly suspect that it's something on your Mac that's at issue. The plugin's working as intended.

Good luck...let me know if you do get it up and running, or if you run into another issue.

Closing this one.

**Comment by fhughes90** [comment]
Sorry @kortstee, really late to respond but just now seeing this. I was able to get it resolved after I started on a clean slate on my RPi. New OS and reinstalled all my services. I want to say I may of had an issue with some library dependancies conflicting each other? I'm not really sure but I know it now works for my macOS.

I did convert my architecture to the Raspian OS Lite 64bit and moved home bridge (and all my other services) into docker. I hope this helps!
